### PR TITLE
fixed #47: Moved REST API usage examples to /docs/usage.md

### LINKED ISSUE #47: Move API documentation to new doc directory
Move all the REST API usage examples from the `README.md` to a new `/docs` directory, under `usage.md`.

The `README.md` is too cluttered at the moment, and needs to be broken into smaller docs that can be linked from this file.

### PR DESCRIPTION
**Pull request by jatinkumarg**
Moved the REST API usage examples from the `README.md` to a `/docs/usage.md`.

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
+### REST API Usage (`api/v1`)
```
Comment (nitishm) [inline docs/usage.md]:
```suggestion
# REST API Usage (`api/v1`)
```

File: docs/usage.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
+### REST API Usage (`api/v1`)
+
+#### Submit an attack - `POST api/v1/attack`
```
Comment (nitishm) [inline docs/usage.md]:
```suggestion
## Submit an attack - `POST api/v1/attack`
```

File: docs/usage.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+      "scheme": "http"
+    }
+  },
+  "created_at": "Mon, 18 Feb 2019 19:48:19 EST",
+  "updated_at": "Mon, 18 Feb 2019 19:48:33 EST"
+}
+```
+*The returned JSON body includes the **Attack ID** (`494f98a2-7165-4d1b-8834-3226b49ab582`) and the **Attack Status** (`scheduled`).*
+
+#### View attack status by **Attack ID** - `GET api/v1/attack/<attackID>`
```
Comment (nitishm) [inline docs/usage.md]:
```suggestion
## View attack status by **Attack ID** - `GET api/v1/attack/<attackID>`
```

File: docs/usage.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      "scheme": "http"
+    }
+  },
+  "created_at": "Mon, 18 Feb 2019 19:48:19 EST",
+  "updated_at": "Mon, 18 Feb 2019 19:48:33 EST"
+
+}
+```
+
+#### List all attacks `GET /api/v1/attack`
```
Comment (nitishm) [inline docs/usage.md]:
```suggestion
## List all attacks `GET /api/v1/attack`
```

File: docs/usage.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+Duration      [total, attack, wait]    1.993288918s, 1.990719s, 2.569918ms
+Latencies     [mean, 50, 95, 99, max]  2.136603ms, 1.642011ms, 4.151042ms, 9.884504ms, 15.338328ms
+Bytes In      [total, mean]            0, 0.00
+Bytes Out     [total, mean]            0, 0.00
+Success       [ratio]                  0.00%
+Status Codes  [code:count]             404:200  
+Error Set:
+404 Not Found
+```
+#### List all attack reports - `GET api/v1/report`
```
Comment (nitishm) [inline docs/usage.md]:
```suggestion
## List all attack reports - `GET api/v1/report`
```

File: docs/usage.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        "updated_at": "Mon, 18 Feb 2019 19:48:33 EST"
+    }
+]
+```
+
+#### View attack report by **Attack ID** - `GET /api/v1/report/<attackID>[?format=json/text/binary]`
+
+> The report endpoint only returns results for **Completed** attacks
+
+- *JSON Format*
```
Comment (nitishm) [inline docs/usage.md]:
```suggestion
### JSON Format
```

File: docs/usage.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    "rate": 5.347450602925056,
+    "success": 1,
+    "status_codes": {
+        "200": 15
+    },
+    "errors": []
+}
+```
+
+- *Text Format*
```
Comment (nitishm) [inline docs/usage.md]:
```suggestion
### Text Format
```

File: docs/usage.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+                "scheme": "http"
+            }
+        },
+        "created_at": "Mon, 18 Feb 2019 19:48:19 EST",
+        "updated_at": "Mon, 18 Feb 2019 19:48:33 EST"
+    }
+]
+```
+
+#### View attack report by **Attack ID** - `GET /api/v1/report/<attackID>[?format=json/text/binary]`
```
Comment (nitishm) [inline docs/usage.md]:
```suggestion
## View attack report by **Attack ID** - `GET /api/v1/report/<attackID>[?format=json/text/binary]`
```

**Comment by jatinkumarg** [comment]
@nitishm Can you please review my new changes?

**Comment by nitishm** [comment]
@jatinkumarg it looks good now. I forgot to ask you to add a link to the docs in the README.md. 

Under the subsection for Usage add a link to docs.md

**Comment by jatinkumarg** [comment]
@nitishm you mean adding a Usage.md link to Readme.md ? 

Can you also tell me how can I do it?

File: README.md [hunk 9]
```diff
@@ -67,231 +67,6 @@ make container_run
 ```
 > NOTE: `make container` and `make container_clean` can be used to build the Dockerfile and delete the container and image.
```
Comment (nitishm) [inline README.md]:
```suggestion

### REST API
- (usage.md)[https://github.com/nitishm/vegeta-server/tree/master/docs/usage.md]
```

**Comment by nitishm** [comment]
> @nitishm you mean adding a Usage.md link to Readme.md ?
> 
> Can you also tell me how can I do it?

Sure, I added the suggestion for change in the `README.md` change review.

**Comment by jatinkumarg** [comment]
@nitishm Thanks for the help!

I have made the required changes. Can you please review it?

File: README.md [hunk 10]
```diff
@@ -67,230 +67,8 @@ make container_run
...
-        "success": 1,
-        "status_codes": {
-            "200": 5
-        },
-        "errors": []
-    }
-]
-```
+### REST API
+- (usage.md)[https://github.com/nitishm/vegeta-server/tree/master/docs/usage.md]
```
Comment (nitishm) [inline README.md]: Bah my bad. Always mix up the braces.

```suggestion
- [usage.md](https://github.com/nitishm/vegeta-server/tree/master/docs/usage.md)
```

**Comment by nitishm** [comment]
@jatinkumarg  Sorry I mixed up the braces. Suggested the fix.

**Comment by jatinkumarg** [comment]
@nitishm I have updated the brackets in README.md

**Comment by nitishm** [comment]
LGTM. Great job @jatinkumarg

**Pull request merged**